### DISCUSSION TITLE
Milo server returns unknown endpoint addresses

### DISCUSSION THREAD

**Discussion by MeranaTona** [eclipse-milo/milo#1034] [General]
Hi all,

I have set up an OPC UA Milo server on K8s using https://github.com/apache/camel/tree/main/components/camel-milo/src/main/java/org/apache/camel/component/milo/server

Basically the server is reachable over opc.tcp://host-ip:k8s-defined-port, which then gets forwarded to the server container.

The problem is, once connecting to the server via UAExpert over opc.tcp://host-ip:k8s-defined-port, the server returns multiple endpoint addresses which are unkown to the outside network and clients. Meaning, the client cant connect to them. See my post here: https://github.com/apache/camel-k/issues/3580#issuecomment-1230478273

 Can I overwrite those addresses on server side?

Most server configs are done here:
https://github.com/apache/camel-k/issues/3580#issue-1353033395
and here:
https://github.com/apache/camel/blob/main/components/camel-milo/src/main/java/org/apache/camel/component/milo/server/MiloServerComponent.java

Greetings

**Comment by kevinherron** [comment]
I’m not familiar with the Camel integration, but I suspect the endpoints are coming from here:
https://github.com/apache/camel/blob/adc87b142dcb5b81347cce4a1a262c65cd2c98bd/components/camel-milo/src/main/java/org/apache/camel/component/milo/server/MiloServerComponent.java#L247-L249

I don’t see any configuration that would allow you to change this.

The Milo Demo Server, for example, allows you to configure these hostnames, and that includes the ability to specify arbitrary hostnames that might not be returned by default, or to be selective about which hostnames are used: https://github.com/digitalpetri/opc-ua-demo-server/blob/c9a41c49920635023910d34e760655c95d1a72f9/src/main/kotlin/com/digitalpetri/opcua/server/DemoServer.kt#L362-L364

It sounds like you would need to open this issue against the Camel integration instead.
